stream/src/main/java/org/apache/kafka/streaming/examples/ProcessorJob.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.kafka.streaming.KafkaStreaming;
+import org.apache.kafka.streaming.processor.Processor;
+import org.apache.kafka.streaming.processor.ProcessorDef;
+import org.apache.kafka.streaming.processor.TopologyBuilder;
+import org.apache.kafka.streaming.processor.ProcessorContext;
+import org.apache.kafka.streaming.StreamingConfig;
+import org.apache.kafka.streaming.state.Entry;
+import org.apache.kafka.streaming.state.InMemoryKeyValueStore;
+import org.apache.kafka.streaming.state.KeyValueIterator;
+import org.apache.kafka.streaming.state.KeyValueStore;
+
+import java.util.Properties;
+
+public class ProcessorJob {
+
+    private static class MyProcessorDef implements ProcessorDef {
+
+        @Override
+        public Processor<String, Integer> instance() {
+            return new Processor<String, Integer>() {
+                private ProcessorContext context;
+                private KeyValueStore<String, Integer> kvStore;
+
+                @Override
+                public void init(ProcessorContext context) {
+                    this.context = context;
+                    this.context.schedule(this, 1000);
+                    this.kvStore = new InMemoryKeyValueStore<>("local-state", context);
+                }
+
+                @Override
+                public void process(String key, Integer value) {
+                    Integer oldValue = this.kvStore.get(key);
+                    if (oldValue == null) {
+                        this.kvStore.put(key, value);
+                    } else {
+                        int newValue = oldValue + value;
+                        this.kvStore.put(key, newValue);
+                    }
+
+                    context.commit();
+                }
+
+                @Override
+                public void punctuate(long streamTime) {
+                    KeyValueIterator<String, Integer> iter = this.kvStore.all();
+
+                    while (iter.hasNext()) {
+                        Entry<String, Integer> entry = iter.next();
+
+                        System.out.println("[" + entry.key() + ", " + entry.value() + "]");
+
+                        context.forward(entry.key(), entry.value());
+                    }
+                }
+
+                @Override
+                public void close() {
+                    this.kvStore.close();
+                }
+            };
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        StreamingConfig config = new StreamingConfig(new Properties());
+        TopologyBuilder builder = new TopologyBuilder();
+
+        builder.addSource("SOURCE", new StringDeserializer(), new IntegerDeserializer(), "topic-source");
+
+        builder.addProcessor("PROCESS", new MyProcessorDef(), null, "SOURCE");
+
+        builder.addSink("SINK", "topic-sink", new StringSerializer(), new IntegerSerializer(), "PROCESS");
+
+        KafkaStreaming streaming = new KafkaStreaming(builder, config);
+        streaming.start();
```

Comment:
When a `Processor` instance is started, the framework calls `init(ProcessorContext)`, but with the most recent changes it is no longer possible for a `Processor` implementation to get the configuration from the `ProcessorContext`. How can one pass in the configuration into the `Processor`? For example, my processor implementation might require several configuration properties to control or alter the default behavior. Using the same configuration sure seemed like a natural way to do this.

One option is to pass the `StreamingConfig` object into the `ProcessorDef` constructor, which would change line 94 to be something like:

    builder.addProcessor("PROCESS", new MyProcessorDef(config), null, "SOURCE");

The `ProcessDef` could then pass the configuration into the constructor of the `Processor` implementation. While that works, it seems like this would then require the `TopologyBuilder` to contain objects that are dependent upon a configuration, and that might not be the same configuration passed into `new KafkaStreaming(builder, config);` (line 98 above).

It sure seems better and far simpler to instead allow `Processor.init(ProcessorContext)` access to the same configuration that KafkaStreaming has when it is initializing the processor. IOW, either change `ProcessorContext` to expose the configuration (as before), or (better yet IMO) add a second parameter to `Processor.init(...)` so it then becomes:

    public interface Processor<K, V> {
        void init(ProcessorContext context, StreamingConfig config);
        ...
    }

Thoughts? Am I missing something more obvious?